gprofiler/utils.py

```diff
@@ -440,3 +463,10 @@ def limit_frequency(limit: int, requested: int, msg_header: str, runtime_logger:
         return limit
 
     return requested
+
+
+def get_hostname() -> str:
+    if hostname is None:
+        raise Exception("hostname not initialized!")
+
+    return hostname
```

Comment:
Do we want to return `unknown` when it fails?

Reply:
Nope, I raise intentionally because it shouldn't be called before we try to initialize the hostname. That's actually an `assert`.

Reply:
I meant that after you initialize it already, the hostname fetching might get an error (which you log). In that case, the hostname would be `unknown`.

Reply:
That's the same discussion as https://github.com/Granulate/gprofiler/pull/98#discussion_r646115821, no?

Reply:
yeah, pretty much